google/cloud/storage/internal/async_token_impl.h

```diff
@@ -0,0 +1,40 @@
+// Copyright 2023 Google LLC
```

Comment:
Do you want to put this in: `google/cloud/storage/internal/async/...` ?

Reply:
Done.